Andrew Fresh wrote on Tue, Apr 12, 2016 at 02:35:45PM -0700:

> The utf-8 thing seems to resolve itself by passing the `--utf8`
> flag to pod2man, so maybe it's an easy fix.
> pod2man --utf8 /usr/libdata/perl5/pod/perlunicook.pod | mandoc | less -R

It looks like pod2man(1) is called all over the place by many
components of various build systems, so trying to pass --utf8
everywhere seems tedious, maybe hopeless to me.  Instead, i propose
to make --utf8 the default on OpenBSD and provide --no-utf8 for
people who know what they are doing and do not want it.

What do you think about the patch below?

Commit it?  Post it to a few more people for review?  Or to tech@?
Or forget it and try to track down all calls to pod2man(1) in all
the build systems instead?

Yours,
  Ingo

--- cpan/podlators/scripts/pod2man.PL.orig	Tue Mar  1 04:33:02 2016
+++ cpan/podlators/scripts/pod2man.PL	Thu Sep  1 10:32:44 2016
@@ -77,7 +77,7 @@
 GetOptions (\%options, 'center|c=s', 'date|d=s', 'errors=s', 'fixed=s',
             'fixedbold=s', 'fixeditalic=s', 'fixedbolditalic=s', 'help|h',
             'lax|l', 'name|n=s', 'nourls', 'official|o', 'quotes|q=s',
-            'release|r=s', 'section|s=s', 'stderr', 'verbose|v', 'utf8|u')
+            'release|r=s', 'section|s=s', 'stderr', 'verbose|v', 'utf8|u!')
     or exit 1;
 pod2usage (0) if $options{help};
 
@@ -123,7 +123,7 @@
 __END__
 
 =for stopwords
-en em --stderr stderr --utf8 UTF-8 overdo markup MT-LEVEL Allbery Solaris
+en em --stderr stderr --no-utf8 UTF-8 overdo markup MT-LEVEL Allbery Solaris
 URL troff troff-specific formatters uppercased Christiansen --nourls UTC
 prepend
 
@@ -137,7 +137,7 @@
     [B<--fixed>=I<font>] [B<--fixedbold>=I<font>] [B<--fixeditalic>=I<font>]
     [B<--fixedbolditalic>=I<font>] [B<--name>=I<name>] [B<--nourls>]
     [B<--official>] [B<--quotes>=I<quotes>] [B<--release>=I<version>]
-    [B<--section>=I<manext>] [B<--stderr>] [B<--utf8>] [B<--verbose>]
+    [B<--section>=I<manext>] [B<--stderr>] [B<--no-utf8>] [B<--verbose>]
     [I<input> [I<output>] ...]
 
 pod2man B<--help>
@@ -323,19 +323,10 @@
 
 =item B<-u>, B<--utf8>
 
-By default, B<pod2man> produces the most conservative possible *roff
-output to try to ensure that it will work with as many different *roff
-implementations as possible.  Many *roff implementations cannot handle
-non-ASCII characters, so this means all non-ASCII characters are converted
-either to a *roff escape sequence that tries to create a properly accented
-character (at least for troff output) or to C<X>.
-
-This option says to instead output literal UTF-8 characters.  If your
-*roff implementation can handle it, this is the best output format to use
-and avoids corruption of documents containing non-ASCII characters.
-However, be warned that *roff source with literal UTF-8 characters is not
-supported by many implementations and may even result in segfaults and
-other bad behavior.
+This option allows B<pod2man> to output literal UTF-8 characters.
+On OpenBSD, it is enabled by default and can be disabled with
+B<--no-utf8>, in which case non-ASCII characters are converted
+either to *roff escape sequences or to C<X>.
 
 Be aware that, when using this option, the input encoding of your POD
 source should be properly declared unless it's US-ASCII.  Pod::Simple will
